### PR TITLE
Created product and dispatch models

### PR DESCRIPTION
**Pull request by darshanrangokiya**

### CODE REVIEW

File: stockManagement/dashboard/models.py [hunk 1]
```diff
@@ -1,3 +1,58 @@
...
+    product_type = models.ManyToManyField(ProductType, related_name="product_standards")
+
+    def __str__(self):
+        return f"{self.standard}"
+    
+class Product(BaseModel):
+    design_name = models.CharField(max_length=255)
+    opening_stock = models.IntegerField()
+    company_name = models.CharField(max_length=255, null=True, blank=True)
+    image = models.FileField(upload_to="uploads/", max_length=255, null=True, blank=True)
```
Comment (HarryCreation) [inline stockManagement/dashboard/models.py:36]: Please create one function and then build the path with instance id or many be with product type.
```
def product_file_path(instance, filename):
   return os.path.join(f"{instance.id}", filename)
```
The above one is for reference but use better structure path.

File: stockManagement/dashboard/models.py [hunk 2]
```diff
@@ -1,3 +1,58 @@
...
+    product_type = models.ForeignKey(ProductType, on_delete=models.CASCADE, null=False, blank=False, related_name="products")
+    product_size = models.ForeignKey(ProductSize, on_delete=models.SET, null=True, blank=True, related_name="products")
+    product_category = models.ForeignKey(ProductCategory, on_delete=models.SET_NULL, null=True, blank=True, related_name="products")
```
Comment (HarryCreation) [inline stockManagement/dashboard/models.py:39]: Use unique `related_name` for all ForeignKey fields

**Review by HarryCreation — CHANGES_REQUESTED** [review]

File: stockManagement/dashboard/models.py (hunk 2 above)
Comment (darshanrangokiya) [reply stockManagement/dashboard/models.py:39]: @HarryCreation As all the foreign keys are referring to different models we can provide the same for them.

File: stockManagement/dashboard/models.py (hunk 1 above)
Comment (darshanrangokiya) [reply stockManagement/dashboard/models.py:36]: @HarryCreation Here the problem is that the first time `instance.id` will be None so the None folder will be created. So do we have any other options here.
Or can we use the uuid here instead?

File: stockManagement/dashboard/models.py [hunk 3]
```diff
@@ -1,58 +1,131 @@
...
+        on_delete=models.SET_NULL,
+        related_name="updated_products",
+    )
 
     def __str__(self):
         return f"{self.design_name}"
 
+
+def product_file_path(instance, filename):
+    return os.path.join("uploads", f"{instance.pk}", filename)
```
Comment (HarryCreation) [inline stockManagement/dashboard/models.py:93]:
```
return os.path.join("uploads", f"{instance.product.id}", filename)
```

**Pull request merged**